### PR TITLE
docs: update changelog for release v3.0.4

### PR DESCRIPTION
**Pull request by justxuewei**
**What this PR does**: 

Updated changelog for release `v3.0.4`. Fixed some typos in the changelog.

**Which issue(s) this PR fixes**: 
Fixes #

**You should pay attention to items below to ensure your pr passes our ci test**
We do not merge pr with ci tests failed

- [ ] All ut passed (run 'go test ./...' in project root)
- [ ] After go-fmt ed , run 'go fmt project' using goland.
- [ ] Golangci-lint passed, run 'sudo golangci-lint run' in project root.
- [ ] Your new-created file needs to have [apache license](https://raw.githubusercontent.com/dubbogo/resources/master/tools/license/license.txt) at the top, like other existed file does.
- [ ] All integration test passed. You can run integration test locally (with docker env). Clone our [dubbo-go-samples](https://github.com/apache/dubbo-go-samples) project and replace the go.mod to your dubbo-go, and run 'sudo sh start_integration_test.sh' at root of samples project root. (M1 Slice is not Support)

Signed-off-by: Xuewei Niu <justxuewei@apache.org>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,32 @@
...
+- [Fix: timeout don't support ms bug](https://github.com/apache/dubbo-go/pull/2105)
+- [Fix: env DUBBO_IP_TO_REGISTRY doesn't work](https://github.com/apache/dubbo-go/pull/2138)
+
+### Enhancements
+
+- [Upgrade Triple to v1.1.9](https://github.com/apache/dubbo-go/pull/2072)
+- [Docs: update style and out-of-date links in readme](https://github.com/apache/dubbo-go/pull/2079)
+- [Upgrade Hessian to v1.11.3](https://github.com/apache/dubbo-go/pull/2087)
+- [Upgrade gopkg.in/yaml.v3 to 3.0.0](https://github.com/apache/dubbo-go/pull/2112)
+- [Upgrade nacos to v2.1.2](https://github.com/apache/dubbo-go/pull/2116)
```
Comment (binbin0325) [inline CHANGELOG.md:27]: 准备降级nacos到v1版本了，这个可以删掉。

Comment (binbin0325) [inline CHANGELOG.md:27]: 准备降级nacos到v1版本了，这个可以删掉。

**Review by zhaoyunxing92 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (justxuewei) [reply CHANGELOG.md]: done

**Comment by justxuewei** [comment]
> pls add #2117

Done.

**Review by binbin0325 — APPROVED** [review]

**Pull request closed without merging**